functional test fails: swapfile detection swapfile created before boot vim-patch:8.2.2586

### Problem

When building neovim-0.11.1 for Alpine Linux I bumped into a test failure:



```
ncopa-edge-x86-64:~/aports/community/neovim/src/neovim-0.11.1 (neovim-0.11)$ TEST_FILE=test/functional/ex_cmds/swapfile_preserve_recover_spec.lua  make functionaltest
...
...
[140/140] cd /home/ncopa/aports/community/neovim/src/neovim-0.11.1/build/test && /usr/bin/cmake -D TEST_TYPE=functional -D BUILD_DIR=/home/ncopa/aports/community/neovim/src/neovim-0.11.1/build -
D CIRRUS_CI= -D CI_BUILD=OFF -D DEPS_INSTALL_DIR=/home/ncopa/aports/community/neovim/src/neovim-0.11.1/build/usr -D NVIM_PRG=/home/ncopa/aports/community/neovim/src/neovim-0.11.1/build/bin/nvim 
-D TEST_DIR=/home/ncopa/aports/community/neovim/src/neovim-0.11.1/test -D WORKING_DIR=/home/ncopa/aports/community/neovim/src/neovim-0.11.1 -P /home/ncopa/aports/community/neovim/src/neovim-0.11
.1/cmake/RunTests.cmake                                                                                                                                                                           
-------- Global test environment setup.                                                                                                                                                           
-------- Running tests from test/functional/ex_cmds/swapfile_preserve_recover_spec.lua                                                                                                            
RUN      T1 :recover fails if given a non-existent swapfile: 13.00 ms OK                                                                                                                          
RUN      T2 preserve and (R)ecover with custom 'directory' with :preserve and SIGKILL: 34.00 ms OK                                                                                                
RUN      T3 preserve and (R)ecover with custom 'directory' closing stdio channel without :preserve #22096: 26.00 ms OK                                                                            
RUN      T4 preserve and (R)ecover with custom 'directory' killing TUI process without :preserve #22096: 1076.00 ms OK                                                                            
RUN      T5 swapfile detection redrawing during prompt does not break treesitter: 1041.00 ms OK                                                                                                   
RUN      T6 swapfile detection always show swapfile dialog #8840 #9027: 87.00 ms OK                                                                                                               
RUN      T7 swapfile detection default SwapExists handler selects "(E)dit" and skips prompt: 41.00 ms OK                                                                                          
RUN      T9 swapfile detection selecting "q" in the attention prompt: 76.00 ms OK                                                                                                                 
RUN      T12 swapfile detection swapfile created before boot vim-patch:8.2.2586: 3530.00 ms FAIL                                                                                                  
test/functional/ex_cmds/swapfile_preserve_recover_spec.lua:450: Failed to match any screen lines.                                                                                                 
Expected (anywhere): "{2:E325: ATTENTION}.*file name: .*Xswaptest.*process ID: %d* %(STILL RUNNING%).*{1:%[O%]pen Read%-Only, %(E%)dit anyway, %(R%)ecover, %(Q%)uit, %(A%)bort: }%^"             
Actual:                                                                                                                                                                                           
  |                                                                           |                                                                                                                   
  |{0:~                                                                          }|                                                                                                               
  |{UNEXPECTED reverse = true, bold = true:                                                                           }|                                                                          
  |{2:E325: ATTENTION}                                                            |                                                                                                               
  |Found a swap file by the name "~/aports/community/neovim/src/neovim-0.11.1/|                                                                                                                   
  |Xtest_swapdialog_dir/%home%ncopa%aports%community%neovim%src%neovim-0.11.1%|                                                                                                                   
  |Xswaptest.swp"                                                             |                                                                                                                   
  |          owned by: ncopa   dated: Wed May 07 18:05:42 2025                |                                                                                                                   
  |         file name: ~ncopa/aports/community/neovim/src/neovim-0.11.1/Xswapt|                                                                                                                   
  |est                                                                        |                                                                                                                   
  |          modified: YES                                                    |                                                                                                                   
  |         user name: ncopa   host name: ncopa-edge-x86-64                   |                                                                                                                   
  |        process ID: 14833 (STILL RUNNING)                                  |                                                                                                                   
  |While opening file "Xswaptest"                                             |                                                                                                                   
  |      CANNOT BE FOUND                                                      |                                                                                                                   
  |(1) Another program may be editing the same file.  If this is the case,    |                                                                                                                   
  |    be careful not to end up with two different instances of the same      |                                                                                                                   
  |    file when making changes.  Quit, or continue with caution.             |                                                                                                                   
  |(2) An edit session for this file crashed.                                 |                                                                                                                   
  |    If this is the case, use ":recover" or "nvim -r Xswaptest"             |                                                                                                                   
  |    to recover the changes (see ":help recovery").                         |                                                                                                                   
  |    If you did this already, delete the swap file "/home/ncopa/aports/commu|                                                                                                                   
  |nity/neovim/src/neovim-0.11.1/Xtest_swapdialog_dir/%home%ncopa%aports%commu|                                                                                                                   
  |nity%neovim%src%neovim-0.11.1%Xswaptest.swp"                               |                                                                                                                   
  |    to avoid this message.                                                 |                                                                                                                   
  |                                                                           |                                                                                                                   
  |{1:Swap file "~/aports/community/neovim/src/neovim-0.11.1/Xtest_swapdialog_dir}|                                                                                                               
  |{1:/%home%ncopa%aports%community%neovim%src%neovim-0.11.1%Xswaptest.swp" alrea}|                                                                                                               
  |{1:dy exists!}                                                                 |                                                                                                               
  |{1:[O]pen Read-Only, (E)dit anyway, (R)ecover, (Q)uit, (A)bort: }^              |                                                                                                              
                                                                                                                                                                                                  
                                                                                                                                                                                                  

Snapshot:                                                                                                                                                                                         
screen:set_default_attr_ids({                                                                                                                                                                     
  [0] = { foreground = Screen.colors.Blue, bold = true },                                                                                                                                         
  [1] = { foreground = Screen.colors.SeaGreen4, bold = true },                                                                                                                                    
  [2] = { background = Screen.colors.Red1, foreground = Screen.colors.Gray100 },                                                                                                                  
  [3] = { reverse = true, bold = true },                                                                                                                                                          
})                                                                                                                                                                                                
                                                                                                                                                                                                  
screen:expect([[                                                                                                                                                                                  
                                                                             |                                                                                                                    
  {0:~                                                                          }|                                                                                                                
  {3:                                                                           }|                                                                                                                
  {2:E325: ATTENTION}                                                            |                                                                                                                
  Found a swap file by the name "~/aports/community/neovim/src/neovim-0.11.1/|                                                                                                                    
  Xtest_swapdialog_dir/%home%ncopa%aports%community%neovim%src%neovim-0.11.1%|                                                                                                                    
  Xswaptest.swp"                                                             |                                                                                                                    
            owned by: ncopa   dated: Wed May 07 18:05:42 2025                |                                                                                                                    
           file name: ~ncopa/aports/community/neovim/src/neovim-0.11.1/Xswapt|                                                                                                                    
  est                                                                        |                                                                                                                    
            modified: YES                                                    |                                                                                                                    
           user name: ncopa   host name: ncopa-edge-x86-64                   |                                                                                                                    
          process ID: 14833 (STILL RUNNING)                                  |                                                                                                                    
  While opening file "Xswaptest"                                             |                                                                                                                    
        CANNOT BE FOUND                                                      |                                                                                                                    
  (1) Another program may be editing the same file.  If this is the case,    |                                                                                                                    
      be careful not to end up with two different instances of the same      |                                                                                                                    
      file when making changes.  Quit, or continue with caution.             |                                                                                                                    
  (2) An edit session for this file crashed.                                 |                                                                                                                    
      If this is the case, use ":recover" or "nvim -r Xswaptest"             |                                                                                                                    
      to recover the changes (see ":help recovery").                         |                                                                                                                    
      If you did this already, delete the swap file "/home/ncopa/aports/commu|                                                                                                                    
  nity/neovim/src/neovim-0.11.1/Xtest_swapdialog_dir/%home%ncopa%aports%commu|                                                                                                                    
  nity%neovim%src%neovim-0.11.1%Xswaptest.swp"                               |                                                                                                                    
      to avoid this message.                                                 |                                                                                                                    
                                                                             |                                                                                                                    
  {1:Swap file "~/aports/community/neovim/src/neovim-0.11.1/Xtest_swapdialog_dir}|                                                                                                                
  {1:/%home%ncopa%aports%community%neovim%src%neovim-0.11.1%Xswaptest.swp" alrea}|                                                                                                                
  {1:dy exists!}                                                                 |                                                                                                                
  {1:[O]pen Read-Only, (E)dit anyway, (R)ecover, (Q)uit, (A)bort: }^              |                                                                                                               
]])                                                                                                                                                                                               
                                                                                                                                                                                                  
stack traceback:                                                                                                                                                                                  
        test/functional/ui/screen.lua:852: in function '_wait'                                                                                                                                    
        test/functional/ui/screen.lua:517: in function 'expect'                                                                                                                                   
        test/functional/ex_cmds/swapfile_preserve_recover_spec.lua:450: in function 'on_swapfile_running'                                                                                         
        test/functional/ex_cmds/swapfile_preserve_recover_spec.lua:421: in function 'test_swapfile_after_reboot'                                                                                  
        test/functional/ex_cmds/swapfile_preserve_recover_spec.lua:449: in function <test/functional/ex_cmds/swapfile_preserve_recover_spec.lua:448>                                              
                                                                                                                                                                                                  
RUN      T13 swapfile detection swapfile created before boot + default SwapExists handler: 64.00 ms OK                                                                                            
RUN      T15 quitting swapfile dialog on startup stops TUI properly (Q)uit at first file argument: 46.00 ms OK                                                                                    
RUN      T17 quitting swapfile dialog on startup stops TUI properly (A)bort at second file argument with -p: 68.00 ms OK                                                                          
RUN      T19 quitting swapfile dialog on startup stops TUI properly (Q)uit at file opened by -t: 51.00 ms OK                                                                                      
-------- 13 tests from test/functional/ex_cmds/swapfile_preserve_recover_spec.lua (6311.00 ms total)                                                                                              
                                                                                                                                                                                                  
-------- Global test environment teardown.                                                                                                                                                        
======== 13 tests from 1 test file ran. (6312.00 ms total)                                                                                                                                        
PASSED   12 tests.
FAILED   1 test, listed below:                                                                                                                                                                    FAILED   test/functional/ex_cmds/swapfile_preserve_recover_spec.lua @ 448: swapfile detection swapfile created before boot vim-patch:8.2.2586                                                     test/functional/ex_cmds/swapfile_preserve_recover_spec.lua:450: Failed to match any screen lines.                                                                                                 Expected (anywhere): "{2:E325: ATTENTION}.*file name: .*Xswaptest.*process ID: %d* %(STILL RUNNING%).*{1:%[O%]pen Read%-Only, %(E%)dit anyway, %(R%)ecover, %(Q%)uit, %(A%)bort: }%^"             
Actual:                                                                                                                   
  |                                                                           |                                           
  |{0:~                                                                          }|                                       
  |{UNEXPECTED reverse = true, bold = true:                                                                           }|
  |{2:E325: ATTENTION}                                                            |                                       
  |Found a swap file by the name "~/aports/community/neovim/src/neovim-0.11.1/|                                           
  |Xtest_swapdialog_dir/%home%ncopa%aports%community%neovim%src%neovim-0.11.1%|                                         
  |Xswaptest.swp"                                                             |                                         
  |          owned by: ncopa   dated: Wed May 07 18:05:42 2025                |     
  |         file name: ~ncopa/aports/community/neovim/src/neovim-0.11.1/Xswapt|     
  |est                                                                        |           
  |          modified: YES                                                    |           
  |         user name: ncopa   host name: ncopa-edge-x86-64                   |               
  |        process ID: 14833 (STILL RUNNING)                                  |               
  |While opening file "Xswaptest"                                             |    
  |      CANNOT BE FOUND                                                      |    
  |(1) Another program may be editing the same file.  If this is the case,    |                                           
  |    be careful not to end up with two different instances of the same      |                                           
  |    file when making changes.  Quit, or continue with caution.             |                                           
  |(2) An edit session for this file crashed.                                 |                                           
  |    If this is the case, use ":recover" or "nvim -r Xswaptest"             |                                           
  |    to recover the changes (see ":help recovery").                         |                                           
  |    If you did this already, delete the swap file "/home/ncopa/aports/commu|                                           
  |nity/neovim/src/neovim-0.11.1/Xtest_swapdialog_dir/%home%ncopa%aports%commu|                                           
  |nity%neovim%src%neovim-0.11.1%Xswaptest.swp"                               |                                                                                                                     |    to avoid this message.                                                 |                                                                                                                     |                                                                           |                                                                                                                     |{1:Swap file "~/aports/community/neovim/src/neovim-0.11.1/Xtest_swapdialog_dir}|                                                                                                               
  |{1:/%home%ncopa%aports%community%neovim%src%neovim-0.11.1%Xswaptest.swp" alrea}|                                       
  |{1:dy exists!}                                                                 |                                                                                                               
  |{1:[O]pen Read-Only, (E)dit anyway, (R)ecover, (Q)uit, (A)bort: }^              |                                                                                                              
                                                                                                                                                                                                  


Snapshot:                                                                                        
screen:set_default_attr_ids({                                                                    
  [0] = { foreground = Screen.colors.Blue, bold = true },                                        
  [1] = { foreground = Screen.colors.SeaGreen4, bold = true },                                                                                                                                    
  [2] = { background = Screen.colors.Red1, foreground = Screen.colors.Gray100 },                                                                                                                  
  [3] = { reverse = true, bold = true },                                                         
})                                                                                               

screen:expect([[                                                                                 
                                                                             |                                                                                                                    
  {0:~                                                                          }|                                                                                                                
  {3:                                                                           }|                                                                                                                
  {2:E325: ATTENTION}                                                            |                                                                                                                
  Found a swap file by the name "~/aports/community/neovim/src/neovim-0.11.1/|                                                                                                                    
  Xtest_swapdialog_dir/%home%ncopa%aports%community%neovim%src%neovim-0.11.1%|                                                                                                                    
  Xswaptest.swp"                                                             |                                                                                                                    
            owned by: ncopa   dated: Wed May 07 18:05:42 2025                |                                                                                                                    
           file name: ~ncopa/aports/community/neovim/src/neovim-0.11.1/Xswapt|                                                                                                                    
  est                                                                        |                                                                                                                    
            modified: YES                                                    |                                                                                                                    
           user name: ncopa   host name: ncopa-edge-x86-64                   |                                                                                                                    
          process ID: 14833 (STILL RUNNING)                                  |                                                                                                                    
  While opening file "Xswaptest"                                             |                                                                                                                    
        CANNOT BE FOUND                                                      |                                                                                                                    
  (1) Another program may be editing the same file.  If this is the case,    |                                                                                                                    
      be careful not to end up with two different instances of the same      |                                                                                                                    
      file when making changes.  Quit, or continue with caution.             |                                                                                                                    
  (2) An edit session for this file crashed.                                 |                                                                                                                    
      If this is the case, use ":recover" or "nvim -r Xswaptest"             |                                                                                                                    
      to recover the changes (see ":help recovery").                         |                                                                                                                    
      If you did this already, delete the swap file "/home/ncopa/aports/commu|                                                                                                                    
  nity/neovim/src/neovim-0.11.1/Xtest_swapdialog_dir/%home%ncopa%aports%commu|                                                                                                                    
  nity%neovim%src%neovim-0.11.1%Xswaptest.swp"                               |                                                                                                                    
      to avoid this message.                                                 |                                                                                                                    
                                                                             |                                                                                                                    
  {1:Swap file "~/aports/community/neovim/src/neovim-0.11.1/Xtest_swapdialog_dir}|                                                                                                                
  {1:/%home%ncopa%aports%community%neovim%src%neovim-0.11.1%Xswaptest.swp" alrea}|                                                                                                                
  {1:dy exists!}                                                                 |                                                                                                                
  {1:[O]pen Read-Only, (E)dit anyway, (R)ecover, (Q)uit, (A)bort: }^              |                                                                                                               
]])                                                                                              

stack traceback:                                                                                 
        test/functional/ui/screen.lua:852: in function '_wait'                                                                                                                                    
        test/functional/ui/screen.lua:517: in function 'expect'                                                                                                                                   
        test/functional/ex_cmds/swapfile_preserve_recover_spec.lua:450: in function 'on_swapfile_running'                                                                                         
        test/functional/ex_cmds/swapfile_preserve_recover_spec.lua:421: in function 'test_swapfile_after_reboot'                                                                                  
        test/functional/ex_cmds/swapfile_preserve_recover_spec.lua:449: in function <test/functional/ex_cmds/swapfile_preserve_recover_spec.lua:448>


 1 FAILED TEST                                                                                   
------------------------------------------------------------------------------                                                                                                                    
$NVIM_LOG_FILE: /home/ncopa/aports/community/neovim/src/neovim-0.11.1/build/.nvimlog                                                                                                              
(last 10 lines)                                                                                  
DBG 2025-05-07T18:05:46.673 T19.14844.0 inbuf_poll:516: blocking... events=true                                                                                                                   
DBG 2025-05-07T18:05:46.683 T19.14844.0 receive_msgpack:210: ch 1: parsing 109 bytes from msgpack Stream: 0x7f02d1d21080                                                                          
DBG 2025-05-07T18:05:46.683 T19.14844.0 RPC: <- 1: [request]   id=6: nvim_eval                                                                                                                    
DBG 2025-05-07T18:05:46.683 T19.14844.0 handle_request:348: RPC: scheduled nvim_eval                                                                                                              
DBG 2025-05-07T18:05:46.683 T19.14844.0 state_enter:99: input: K_EVENT                                                                                                                            
DBG 2025-05-07T18:05:46.683 T19.14844.0 handle_nvim_eval:9682: RPC: ch 1: invoke nvim_eval                                                                                                        
DBG 2025-05-07T18:05:46.683 T19.14844.0 RPC: -> 1: [response]  id=6                                                                                                                               
DBG 2025-05-07T18:05:46.683 T19.14844.0 may_trigger_safestate:307: SafeState: Start triggering                                                                                                    
DBG 2025-05-07T18:05:46.683 T19.14844.0 inbuf_poll:516: blocking... events=false                                                                                                                  
DBG 2025-05-07T18:05:46.683 T19.14844.0 inbuf_poll:516: blocking... events=true                                                                                                                   
------------------------------------------------------------------------------                                                                                                                    
-- Tests exited non-zero: 1                                                                      
-- Output to stderr:                                                                             
E5113: Error while calling lua chunk:                                                            

CMake Error at /home/ncopa/aports/community/neovim/src/neovim-0.11.1/cmake/RunTests.cmake:104 (message):                                                                                          
  functional tests failed with error: 1                                                          


ninja: job failed: cd /home/ncopa/aports/community/neovim/src/neovim-0.11.1/build/test && /usr/bin/cmake -D TEST_TYPE=functional -D BUILD_DIR=/home/ncopa/aports/community/neovim/src/neovim-0.11.
1/build -D CIRRUS_CI= -D CI_BUILD=OFF -D DEPS_INSTALL_DIR=/home/ncopa/aports/community/neovim/src/neovim-0.11.1/build/usr -D NVIM_PRG=/home/ncopa/aports/community/neovim/src/neovim-0.11.1/build/
bin/nvim -D TEST_DIR=/home/ncopa/aports/community/neovim/src/neovim-0.11.1/test -D WORKING_DIR=/home/ncopa/aports/community/neovim/src/neovim-0.11.1 -P /home/ncopa/aports/community/neovim/src/ne
ovim-0.11.1/cmake/RunTests.cmake
ninja: subcommand failed                                                                         

make: *** [Makefile:142: functionaltest] Error 1
```

### Steps to reproduce

build neovim-0.11.1 on alpine linux edge.

```sh
build() {
	local cmakeoptions=
	case "$CARCH" in
		ppc64le | riscv64 | s390x | loongarch64) cmakeoptions="-DPREFER_LUA=ON" ;;
	esac

	# DCI_BUILD: See https://github.com/neovim/neovim/issues/22932
	cmake -B build -G Ninja \
		-DCMAKE_BUILD_TYPE=MinSizeRel \
		-DCMAKE_INSTALL_PREFIX=/usr \
		-DCMAKE_INSTALL_LIBDIR=lib \
		-DENABLE_LTO=TRUE \
		-DCMAKE_VERBOSE_MAKEFILE=TRUE \
		-DCI_BUILD=OFF \
		-DENABLE_TRANSLATIONS=ON \
		$cmakeoptions
	cmake --build build
}

check() {
	# Required by the unit tests but not set on the Alpine GitLab CI.
	export USER="$(id -un)"

	# Without this, the temp directory for the tests exceeds the sun_path
	# buffer in sockaddr_un, causing test failures due to path trunctation.
	export XDG_RUNTIME_DIR=/tmp

	# Needed to ensure that neovim finds the tree-parser grammars.
	ln -Tfs /usr/lib/tree-sitter "$builddir"/runtime/parser

	export TEST_TIMEOUT=12000

	msg "Running unit tests"
	cmake --build build --target unittest

	# These tests fail due to terminal attribute difference. There have been
	# some changes in this regard on the neovim master. If the tests continue
	# to fail with neovim >= 0.11.0, then this should be reported upstream.
	#export TEST_FILTER_OUT='.*swapfile.*'

	# The functional tests are known to fail on 32-bit arches.
	# See: https://github.com/neovim/neovim/issues/32230
	#case "$CARCH" in
	#arm*|x86) return ;;
	#esac

	msg "Running functional tests"
	cmake --build build --target functionaltest
}
```

### Expected behavior

functional tests passes.

### Nvim version (nvim -v)

0.11.1

### Vim (not Nvim) behaves the same?

does not run lua function tests

### Operating system/version

alpine linux edge

### Terminal name/version

tmux

### $TERM environment variable

tmux-256color

### Installation

tried to build neovim package for Alpine Linux

--- a/test/functional/ex_cmds/swapfile_preserve_recover_spec.lua
+++ b/test/functional/ex_cmds/swapfile_preserve_recover_spec.lua
@@ -451,8 +451,8 @@ pcall(vim.cmd.edit, 'Xtest_swapredraw.lua')
       screen:expect({
         any = table.concat({
           pesc('{2:E325: ATTENTION}'),
-          'file name: .*Xswaptest',
-          'process ID: %d* %(STILL RUNNING%)',
+          '\n        process ID: %d* %(STILL RUNNING%)',
+          '\nWhile opening file "Xswaptest"',
           pesc('{1:[O]pen Read-Only, (E)dit anyway, (R)ecover, (Q)uit, (A)bort: }^'),
         }, '.*'),
       })

